src/AppBundle/Entity/CommitteeMembership.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+
+/**
+ * This entity represents a committee membership.
+ *
+ * Each instance of CommitteeMembership means an Adherent is a member of a Committee.
+ *
+ * @ORM\Table(
+ *   name="committees_memberships",
+ *   uniqueConstraints={
+ *     @ORM\UniqueConstraint(
+ *       name="adherent_has_joined_committee",
+ *       columns={"adherent_uuid", "committee_uuid"}
+ *     )
+ *   },
+ *   indexes={
+ *     @ORM\Index(name="committees_memberships_role_idx", columns="privilege")
+ *   }
+ * )
+ * @ORM\Entity(repositoryClass="AppBundle\Repository\CommitteeMembershipRepository")
+ */
+class CommitteeMembership
+{
+    const COMMITTEE_HOST = 'HOST';
+    const COMMITTEE_FOLLOWER = 'FOLLOWER';
+
+    use EntityIdentityTrait;
+
+    /**
+     * The committee UUID.
+     *
+     * @var UuidInterface
+     *
+     * @ORM\Column(type="uuid")
+     */
+    private $committeeUuid;
+
+    /**
+     * The adherent UUID.
+     *
+     * @var UuidInterface
+     *
+     * @ORM\Column(type="uuid")
+     */
+    private $adherentUuid;
+
+    /**
+     * The privilege given to the member in the committee.
+     *
+     * Privilege is either HOST or FOLLOWER.
+     *
+     * @var string
+     *
+     * @ORM\Column(length=10)
+     */
+    private $privilege;
+
+    /**
+     * The date and time when the adherent joined the committee.
+     *
+     * @var \DateTime
+     *
+     * @ORM\Column(type="datetime")
+     */
+    private $joinedAt;
+
+    /**
+     * Constructor.
+     *
+     * @param UuidInterface $uuid
+     * @param UuidInterface $committeeUuid
+     * @param UuidInterface $adherentUuid
+     * @param string        $privilege
+     * @param string        $joinedAt
+     */
+    private function __construct(
+        UuidInterface $uuid,
+        UuidInterface $committeeUuid,
+        UuidInterface $adherentUuid,
+        string $privilege = self::COMMITTEE_FOLLOWER,
+        string $joinedAt = 'now'
+    ) {
+        $this->uuid = $uuid;
+        $this->committeeUuid = $committeeUuid;
+        $this->adherentUuid = $adherentUuid;
+        $this->privilege = $privilege;
+        $this->joinedAt = new \DateTime($joinedAt);
+    }
+
+    /**
+     * Returns a new host membership for an adherent on a committee.
+     *
+     * @param UuidInterface $committeeUuid
+     * @param UuidInterface $adherentUuid
+     * @param string        $joinedAt
+     *
+     * @return CommitteeMembership
+     */
+    public static function createWithHostPrivilege(
```

Comment:
Not here because this method is used in the `Committee` entity class so no way to access a factory object.